tutorials/camkes-vm-kernel-module/camkes-vm-kernel-module.md

```diff
@@ -0,0 +1,119 @@
+<!--
+  Copyright 2021 Michael Neises
+
+  SPDX-License-Identifier: BSD-2-Clause
+-->
+# Cross-Compiling Kernel Modules for use with seL4's Linux VM using qemu-arm-virt
+
+This is a procedure to prepare a camkes app for cross-compiling linux kernel modules for use in virtualization.
+
+See the "module_minimal" app for a ready-to-build camkes application.
+
+1. download linux kernel repo
+```
+git clone git://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable.git
+cd linux-stable
+git checkout linux-4.9.y
+git fetch
+```
+
+2. grab the config from seL4
+```
+make clean
+rm Module.symvers
+rm .config
+rm .config.old
+cp .../projects/camkes-vm-images/qemu-arm-virt/linux_configs/config ./.config
+```
+
+3. prepare kernel for cross-compilation, choosing default config options always
+```
+make ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu- prepare
+```
+
+4. build linux kernel (where X is the number of tasks you want to create)
+```
+make -jX ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu-
+```
+
+5. add the newly built kernel to the camkes build path
+```
+cp arch/arm64/boot/Image [...]/projects/camkes-vm-images/qemu-arm-virt/newLinux
+```
+
+6. edit your cmakelists.txt to use the new kernel by replacing this line:
```

Comment:
There are multiple CMakeLists.txt files in the camkes VM project. I assume that this is referring to the tutorial application's CMakeLists.txt, but for someone new who isn't as familiar with the application structure they can be confused. Can you add something explicit here to point to the right file?

Reply:
I've added greater detail here, and at all other steps. I would like to include a permalink to a line in an appropriate file (like the tutorials/camkes-vm-kernel-module/module_minimal/CMakeLists.txt file I've included here), but I'm not sure how to do that without the file already being hosted.

Reply:
That'd be fine -- if you can predict the github URL correctly (e.g. by example from another file in the repo), we can override the link checker for this. The link will become valid when the PR is merged.